### ISSUE TITLE
Doodle Dash codelab needs to be updated for Flame 1.8.0

### ISSUE THREAD

**Issue by domesticmouse** [flutter/flutter#127779]
```console
Analyzing finished_game...

   info - lib/game/doodle_dash.dart:56:9 - 'camera' is deprecated and shouldn't be used. In the future (maybe as early as v1.9.0) this camera will be removed,
              please use the CameraComponent instead.
              
              This is the simplest way of using the CameraComponent:
              1. Add variables for a CameraComponent and a World to your game class
              
                 final world = World();
                 late final CameraComponent cameraComponent;
              
              2. In your `onLoad` method, initialize the cameraComponent and add the world
                 to it.
                 
                 @override
                 void onLoad() {
                   cameraComponent = CameraComponent(world: world);
                   addAll([cameraComponent, world]);
                 }
                 
              3. Instead of adding the root components directly to your game with `add`,
                 add them to the world.
                 
                 world.add(yourComponent);
              
              4. (Optional) If you want to add a HUD component, instead of using
                 PositionType, add the component as a child of the viewport.
                 
                 cameraComponent.viewport.add(yourHudComponent);. Try replacing the use of the deprecated member with the replacement. - deprecated_member_use
```

**Comment by domesticmouse** [comment]
FYI @miquelbeltran

**Comment by miquelbeltran** [comment]
I will see if I can fix this on https://github.com/flutter/codelabs/pull/1634

**Comment by miquelbeltran** [comment]
Scratch that, the changes on Camera/World for 1.8.0 are more complex than what I thought, so that will have to come in another PR

**Comment by domesticmouse** [comment]
> Scratch that, the changes on Camera/World for 1.8.0 are more complex than what I thought, so that will have to come in another PR

Do we want to publish the dash codelab with flame pinned at 1.7.3 while you refactor for flame 1.8.0?

**Comment by domesticmouse** [comment]
This codelab update is held up on internal issues.

Googlers: b/284553092

**Comment by domesticmouse** [comment]
Hey @khanhnwin do you have bandwidth to look at this codelab update? We will need to update the codelab for Flame 1.8.0 fairly soon to keep the codelab up to date.

**Comment by romarito** [comment]
The announcement of Flutter 3.16 highlights game development, but this update remains pending and increasingly outdated, even though it is the only Flutter game codelab. Unfortunate. Totally out of step with the vision of the future that the speeches convey.

**Comment by domesticmouse** [comment]
@romarito Yes, I agree that the out of date code lab should be replaced with an up to date codelab. In fact, if you want to have a sneak peak at something in progress, have a look at https://github.com/flutter/codelabs/tree/main/brick_breaker

**Comment by domesticmouse** [comment]
Replacement codelab released: https://codelabs.developers.google.com/codelabs/flutter-flame-brick-breaker
